addons/account_payment/models/payment_transaction.py

```diff
@@ -196,15 +196,17 @@ def _log_message_on_linked_documents(self, message):
         :return: None
         """
         self.ensure_one()
-        self = self.with_user(SUPERUSER_ID)  # Log messages as 'OdooBot'
+        def get_author():
+            return self.env.user.partner_id.id if self.env.uid == SUPERUSER_ID else self.partner_id.id
+
         if self.source_transaction_id:
             for invoice in self.source_transaction_id.invoice_ids:
-                invoice.message_post(body=message)
+                invoice.message_post(body=message, author_id=get_author())
```

Comment:
```suggestion
                invoice.message_post(body=message, author_id=author.id)
```
Same for calls below.